packages/Ilcsoftpackage/package.py

```diff
@@ -7,6 +7,69 @@
 
 import os
 
+import llnl.util.tty as tty
+import spack.cmd
+import spack.cmd.common.arguments as arguments
+import spack.environment as ev
+import spack.util.environment
+from  spack.util.environment import *
+import spack.user_environment as uenv
+import spack.store
+import os
+
+
+
+
+
+
+def k4_generate_setup_script(env_mod, shell='sh'):
+    """Return shell code corresponding to a EnvironmentModifications object.
+    Contrary to the spack environment_modifications() method, this does not evaluate
+    the current environment, but generates shell code like:
+    export PATH=/new/path:$PATH
+    instead of:
+    export PATH=/new/path:/current/contents/of/PATH;
+    if `/new/path` is to be prepended.
+
+    :param env_mod: spack EnvironmentModifications object
+    :type env_mod: class: `spack.EnvironmentModifications`
+    :return: Shell code corresponding to the environment modifications.
+    :rtype: str
+    """
+    # first, deduplicate paths # TODO
+    #modifications = self.group_by_name()
+    #for name, actions in sorted(modifications.items()):
+    #  self.prune_duplicate_paths(name)
+    # second, keep track if the paths should be set or prepended
+    modifications = env_mod.group_by_name()
+    new_env = {}
+    env_set_not_prepend = {}
+    for name, actions in sorted(modifications.items()):
+        for x in actions:
+            if isinstance(x, SetPath) or isinstance(x, SetEnv):
+                env_set_not_prepend[name] = True
```

Comment:
That's an interesting point. The assumption is that if there is a `SetEnv` action for a path, there should only be this one action. I'd consider it a bug of the package recipe if it tries to append to or set a path that has already been set. (Some packages like pythia8 also don't parse `:` in environment variables, otherwise I could have just skipped this part of the logic) But maybe there is a way to catch this error if it occurs.

Reply:
There was a mistake in my comment. It will take the value of the last action, depending if this is a specific instance or not, so if there are multiple actions it just takes the last one. Though I don't know what modifications or actions are anyway :)